dali/pipeline/data/types.h

```diff
@@ -324,37 +328,41 @@ DLL_PUBLIC inline bool IsValidType(const TypeInfo &type) {
 // the type as a string. This does not work for non-fundamental types,
 // as we do not have any mechanism for calling the constructor of the
 // type when the buffer allocates the memory.
-#define DALI_REGISTER_TYPE(Type, dtype)                             \
+#define DALI_REGISTER_TYPE(Type, TypeName, dtype)                   \
   template <> DLL_PUBLIC string TypeTable::GetTypeName<Type>()      \
-    DALI_TYPENAME_REGISTERER(Type);                                 \
+    DALI_TYPENAME_REGISTERER(TypeName);                                 \
   template <> DLL_PUBLIC DALIDataType TypeTable::GetTypeID<Type>()  \
     DALI_TYPEID_REGISTERER(Type, dtype);                            \
   DALI_STATIC_TYPE_MAPPING(Type, dtype);
 
 // Instantiate some basic types
-DALI_REGISTER_TYPE(NoType, DALI_NO_TYPE);
-DALI_REGISTER_TYPE(uint8, DALI_UINT8);
-DALI_REGISTER_TYPE(int16, DALI_INT16);
-DALI_REGISTER_TYPE(int32, DALI_INT32);
-DALI_REGISTER_TYPE(int64, DALI_INT64);
-DALI_REGISTER_TYPE(float16, DALI_FLOAT16);
-DALI_REGISTER_TYPE(float, DALI_FLOAT);
-DALI_REGISTER_TYPE(double, DALI_FLOAT64);
-DALI_REGISTER_TYPE(bool, DALI_BOOL);
-DALI_REGISTER_TYPE(string, DALI_STRING);
-DALI_REGISTER_TYPE(DALIImageType, DALI_IMAGE_TYPE);
-DALI_REGISTER_TYPE(DALIDataType, DALI_DATA_TYPE);
-DALI_REGISTER_TYPE(DALIInterpType, DALI_INTERP_TYPE);
-DALI_REGISTER_TYPE(DALITensorLayout, DALI_TENSOR_LAYOUT);
+DALI_REGISTER_TYPE(NoType,            NoType,           DALI_NO_TYPE);
```

Comment:
Why? You just made the code more prone to future errors that may result from incomplete or misspelled copy-paste-replace.